File: pyg_pay_interface/src/main/java/com/pyg/pay/service/PayService.java
package com.pyg.pay.service;

import java.util.Map;

/**
 * @author Technicolor
 * @version v1.0
 * @date 2019/4/10 23:01
 * @description TODO
 **/


public interface PayService {


    /**
     * 生成二维码
     * @param out_trade_no
     * @param userId
     * @return
     */
    Map createNative(String out_trade_no, String userId);

    /**
     * 查询订单
     * @param out_trade_no
     * @return
     */
    Map orderQuery(String out_trade_no);

    /**
     * 修改订单状态
     * @param out_trade_no
     * @param transaction_id
     * @param userId
     * @return
     */
    void updateOrder(String out_trade_no, String transaction_id, String userId);
}

File: pyg_seckill_web/src/main/webapp/js/service/payService.js
app.service("payService",function ($http) {



    //生成二维码
    this.createNative=function (out_trade_no) {
        return $http.get("./pay/createNative?out_trade_no="+out_trade_no);
    };
    //查询订单
    this.orderQuery=function (out_trade_no) {
        return $http.get("./pay/orderQuery?out_trade_no="+out_trade_no);
    };

    //修改订单状态
    this.updateOrder=function (out_trade_no,transaction_id) {
        return $http.get("./pay/updateOrder?out_trade_no="+out_trade_no+"&transaction_id="+transaction_id);
    };
});